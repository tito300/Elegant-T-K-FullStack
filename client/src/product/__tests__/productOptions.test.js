import { shallow } from 'enzyme';
import React from 'react';
import ProductOptions from '../productOptions';

let wrapper;
let variants =  {
    male: [
        {
            color: 'red', sizes: [
                { size: 'l', price: 29.99 },
                { size: 'm', price: 25.99 },
            ]
        },
        {
            color: 'blue', sizes: [
                { size: 's', price: 29.99 },
                { size: 'xl', price: 25.99 },
            ]
        }
    ],
    female: [
        {
            color: 'green', sizes: [
                { size: 'xxl', price: 29.99 },
                { size: 'xs', price: 25.99 },
            ]
        },
        {
            color: 'blue', sizes: [
                { size: 'm', price: 29.99 },
                { size: 'l', price: 25.99 },
            ]
        }
    ],
}

describe('ProductImage Component', ()=> {
    beforeAll(()=>{
        let props = {
            variants, 
            handleOptions: jest.fn(), 
            color: null, 
            size: 'l', 
            gender: 'male',
        }
        wrapper = shallow(<ProductOptions {...props} />);
    })
    it('should match snapshot', ()=>{
        expect(wrapper).toMatchSnapshot();
    });
    it('should render "no variants" message', ()=>{
        wrapper.setProps({
            variants: {
                male: [],
                female: [],
            }
        })
        expect(wrapper.find('.optionsContainer').children().length).toBe(1);
    })
    it('should display size options when color is picked', () => {
        wrapper.setProps({
            variants,
            handleOptions: jest.fn(), 
            color: 'red', 
            size: 'l', 
            gender: 'male',
        })
        expect(wrapper.find('.selectSize').exists()).toBeTruthy();
    })
})