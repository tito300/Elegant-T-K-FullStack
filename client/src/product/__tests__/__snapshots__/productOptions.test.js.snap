// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Component should match snapshot 1`] = `
<div
  className="optionsContainer"
>
  <div>
    <p
      className="optionsTitle"
    >
      Gender
    </p>
    <select
      className="selectGender"
      onChange={[MockFunction]}
    >
      <option
        onChange={[MockFunction]}
        value="male"
      >
        Male
      </option>
      <option
        onChange={[MockFunction]}
        value="female"
      >
        Female
      </option>
      <option
        value=""
      >
        none
      </option>
    </select>
  </div>
  <div>
    <p
      className="optionsTitle"
    >
      Color
    </p>
    <div
      className="colorsContainer"
    >
      <span
        className="colorOption"
        id="red"
        key="red"
        onClick={[MockFunction]}
        title="color: red"
      />
      <span
        className="colorOption"
        id="blue"
        key="blue"
        onClick={[MockFunction]}
        title="color: blue"
      />
    </div>
  </div>
</div>
`;
